pkg/solidity-utils/test/FixedPoint.test.ts

```diff
@@ -0,0 +1,45 @@
+import { expect } from 'chai';
+
+import { Contract } from 'ethers';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+
+describe('FixedPoint', () => {
+  let lib: Contract;
+
+  sharedBeforeEach('deploy lib', async () => {
+    lib = await deploy('FixedPointMock', { args: [] });
+  });
+
+  const checkPow = async (x: number, pow: number) => {
+    const result = fp(x ** pow);
+    expect(await lib.powDown(fp(x), fp(pow))).to.equal(result);
+    expect(await lib.powUp(fp(x), fp(pow))).to.equal(result);
+  };
+
+  const checkPows = async (pow: number) => {
+    it('handles small numbers', async () => {
+      await checkPow(0.0007, pow);
+    });
+
+    it('handles medium numbers', async () => {
+      await checkPow(15, pow);
+    });
+
+    it('handles big numbers', async () => {
+      await checkPow(158300, pow);
```

Comment:
```suggestion
      await checkPow(1583156787139, pow);
```

Reply:
This one fails because it loses precision on mulUp when dividing.